actions/new.action.ts

```diff
@@ -674,33 +654,63 @@ const copyEnvFile = async (dir: string, envExample: string, envFile: string) =>
 };
 
 const createRegistry = async (
-  dir: string,
-  shouldInitializePrisma: boolean,
-  shouldInitializeUserService: boolean,
-  shouldInstallMonitoring: boolean,
-  shouldInitializeMonitoring: boolean,
-  shouldInitializeTemporal: boolean,
-  shouldInitializeLogging: boolean,
-  shouldInitializeFileUpload: boolean
+  projectDirectory: string,
+  spec: {
+    projectName: string,
+    prismaSetup: boolean,
+    userServiceSetup: boolean,
+    monitoringSetup: boolean,
+    monitoringServicesSetup: boolean,
+    temporalSetup: boolean,
+    loggingSetup: boolean,
+    fileUploadSetup: boolean,
+    packageManager: string
+  }
 ): Promise<void> => {
-  const filePath = join(process.cwd(), dir, '.stencil');
-  
-  const setupInfo = [
-    shouldInitializePrisma ? 'Prisma Setup' : '',
-    shouldInitializeUserService ? 'User Services Setup' : '',
-    shouldInstallMonitoring ? 'Monitoring Installed' : '',
-    shouldInitializeMonitoring ? 'Monitoring Setup' : '',
-    shouldInitializeTemporal ? 'Temporal Setup' : '',
-    shouldInitializeLogging ? 'Logging Setup' : '',
-    shouldInitializeFileUpload ? 'File Upload Setup' : ''
-  ].filter(info => info !== '').join('\n');
+  const { projectName, packageManager, ...setupInfo } = spec;
 
+  const filePath = join(process.cwd(), projectDirectory, 'spec.yaml');
+  
   try {
     await fs.promises.access(filePath, fs.constants.F_OK);
-    console.log('.stencil file already exists');
   } catch (error) {
-    await fs.promises.writeFile(filePath, setupInfo);
-    console.log('.stencil file created');
+    const tooling = [];
+    if (setupInfo.prismaSetup) tooling.push('prisma');
+    if (setupInfo.userServiceSetup) tooling.push('userService');
+    if (setupInfo.monitoringSetup) tooling.push('monitoring');
+    if (setupInfo.monitoringServicesSetup) tooling.push('monitoringService');
+    if (setupInfo.temporalSetup) tooling.push('temporal');
+    if (setupInfo.loggingSetup) tooling.push('logging');
+    if (setupInfo.fileUploadSetup) tooling.push('fileUpload');
+
+    const yamlContent = yaml.dump({
+      stencil: '0.0.1',
```

Comment:
Why is the version hardcoded? What if we want to update the version or specify an older version of stencil to use?

Reply:
Wanted to know from where should I retrieve this?
The stencil package version, right? (This is where the spec.yaml file is being created when `stencil new` command is run)